test/prototype/test_bitpacking.py

```diff
@@ -1,70 +1,45 @@
-    assert(unpadded.allclose(test_tensor))
+def test_trinary_to_uint8_CPU():
+    test_tensor = torch.randint(-1, 1, (4, 4, 4), dtype=torch.int32)
+    for i in range(len(test_tensor.shape)):
+        packed = pack(test_tensor, "trinary", dimension = i, container_dtype = torch.uint8, device='cpu')
+        unpacked = unpack(packed, "trinary", dimension = i, device='cpu')
+        assert(unpacked.to(torch.int32).allclose(test_tensor))
 
-def test_uint3_to_int16_col_wise_cpu():
-    test_tensor = torch.randint(0, 7, (8, 5), dtype=torch.int16)
-    packed = pack(test_tensor,16, 3, False, device='cpu')
-    unpacked = unpack(packed, 3, False, device='cpu')
-    unpadded = unpacked[:test_tensor.shape[0], ...]
-    assert(unpadded.allclose(test_tensor))
-    
+def test_to_uint8_CPU():
+    for dtype in {torch.uint2, torch.uint3, torch.uint4, torch.uint5, torch.uint6, torch.uint7}:
+        test_tensor = torch.randint(0, 2**dtype_to_bits(dtype), (4, 4, 4), dtype=torch.uint8)
+        for i in range(len(test_tensor.shape)):
+            packed = pack(test_tensor, dtype, dimension = i, container_dtype = torch.uint8, device='cpu')
+            unpacked = unpack(packed, dtype, dimension = i, device='cpu')
+            assert unpacked.allclose(test_tensor), f"Failed for {dtype} on dim {i}"
+            
 @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
-def test_uint4_to_uint8():
-    test_tensor = torch.randint(0, 15, (4, 4), dtype=torch.uint8).cuda()
-    packed = pack(test_tensor, 8, 4)
-    unpacked = unpack(packed, 4)
-    unpadded = unpacked[:test_tensor.shape[0], ...]
-    assert(unpadded.allclose(test_tensor))
-     
-@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
-@pytest.mark.skipif(not has_triton(), reason="unsupported without triton")
-def test_uint4_to_uint8_compile():
-    torch._dynamo.config.specialize_int = True
-    pack_compiled = torch.compile(pack, fullgraph=True)
-    unpack_compiled = torch.compile(unpack, fullgraph=True)
-    test_tensor = torch.randint(0, 15, (3, 4), dtype=torch.uint8).cuda()
-    packed = pack_compiled(test_tensor, 8, 4)
-    unpacked = unpack_compiled(packed, 4)
-    unpadded = unpacked[:test_tensor.shape[0], ...]
-    assert(unpadded.allclose(test_tensor))
-    
-@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
-def test_uint3_to_int16():
-    test_tensor = torch.randint(0, 7, (5, 8), dtype=torch.int16).cuda()
-    packed = pack(test_tensor,16, 3)
-    unpacked = unpack(packed, 3)
-    unpadded = unpacked[:test_tensor.shape[0], ...]
-    assert(unpadded.allclose(test_tensor))
-
-@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
-@pytest.mark.skipif(not has_triton(), reason="unsupported without triton")
-def test_uint2_to_uint8_col_wise_compile():
-    torch._dynamo.config.specialize_int = True
-    pack_compiled = torch.compile(pack, fullgraph=True)
```

Comment:
I noticed you're not using compile anymore? I also worry that functions that include `.shape` in them will cause a cuda sync and be very slow